crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -665,7 +665,14 @@ where
             Ok(outcome) => {
                 match outcome {
                     CanonicalOutcome::AlreadyCanonical { ref header } => {
-                        if cfg!(feature = "optimism") && self.chain_spec().optimism {
+                        let optimism = cfg_if::cfg_if! {
+                            if #[cfg(feautre = "optimism")] {
+                                self.chain_spec().optimism
+                            } else {
+                                false
+                            }
+                        };
+                        if optimism {
```

Comment:
I think this has to be:

```suggestion
                        cfg_if::cfg_if! {
                            if #[cfg(feautre = "optimism")] {
                                let optimism = self.chain_spec().optimism;
                            } else {
                                let optimism = false;
                            }
                        };
                        if optimism {
```